app/src/main/java/com/lightricks/feedexercise/network/DataTransferObjects.kt

```diff
@@ -1,5 +1,32 @@
 package com.lightricks.feedexercise.network
 
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+
 /**
  * todo: add Data Transfer Object data class(es) here
  */
+
+@JsonClass(generateAdapter = true)
+data class FeedItemDto(
+    @Json(name = "configuration")
+    val configuration: String,
+    @Json(name = "id")
+    val id: String,
+    @Json(name = "isNew")
+    val isNew: Boolean,
+    @Json(name = "isPremium")
+    val isPremium: Boolean,
+    @Json(name = "templateCategories")
+    val templateCategories: List<String>,
+    @Json(name = "templateName")
+    val templateName: String,
+    @Json(name = "templateThumbnailURI")
+    val templateThumbnailURI: String
+) {
+    data class FeedItemDtoList(val templatesMetadata: List<FeedItemDto>)
```

Comment:
You have to annotate this class's properties as well.

Reply:
About the class structure it should be the opposite, the FeedItemDto to be in the FeedItemDtoList.

```kotlin
data class FeedItemDtoList(/**....*/) {
    data class Item(/**....*/)
}
```